collectors/python.d.plugin/dns_query_time/dns_query_time.chart.py

```diff
@@ -89,13 +84,15 @@ def dns_req(ns, t, q):
         request = dns.message.make_query(domain, dns.rdatatype.A)
 
         try:
-            dns_start = time()
-            dns.query.udp(request, ns, timeout=t)
-            dns_end = time()
-            query_time = round((dns_end - dns_start) * 1000)
-            q.put({'_'.join(['ns', ns.replace('.', '_')]): query_time})
+            resp = dns.query.udp(request, ns, timeout=t)
+            if not (resp.rcode == dns.rcode.NOERROR or resp.answer):
```

Comment:
`or` ? i recall you said it was `resp.rcode == dns.rcode.NOERROR` and `not resp.answer` in your case?

if so we should use `and`

Reply:
Good catch!